### PR TITLE
Add flag for setting number of service account workers

### PR DESCRIPTION
**Pull request by andybradshaw**
**What this PR does / why we need it**: This PR makes the number of workers for the Service Account controller configurable. We are running into some issues with this controller keeping up, and needed a way of updating the currently hard-coded worker value of 1.

**Special notes for your reviewer**: The documentation for the `--concurrent-serviceaccount-token-syncs` flag seems to indicate that it defaults to 5, but was unable to find anything that handled the default values for the SAControllerConfiguration struct. Please let me know if defaults for that struct are handled in a different way.

**Release note**:

```release-note
Added a configuration flag to controller-manager for number of service account sync workers.
```

### CODE REVIEW

**Comment by andybradshaw** [comment]
I believe I've associated my github account to my CNCF now... let me know if something still needs to be done there

File: pkg/controller/apis/config/v1alpha1/defaults.go
Comment (cheftako) [inline pkg/controller/apis/config/v1alpha1/defaults.go]: It seems this is already being set in SetDefaults_KubeControllerManagerConfiguration above.While the defaults seem to be the same do we want to allow for competing defaults?

**Review by cheftako — APPROVED** [review]
/lgtm

**Comment by andybradshaw** [comment]
@cheftako anything more I need to do here? Wasn't sure if I needed to address the `needs-kind` label.

**Comment by cheftako** [comment]
@andybradshaw At this point you need an approver. I would check the OWNERS files detailed above.

File: api/api-rules/violation_exceptions.list
Comment (lavalamp) [inline api/api-rules/violation_exceptions.list]: Please fix.

File: staging/src/k8s.io/kube-controller-manager/config/v1alpha1/types.go
Comment (lavalamp) [inline staging/src/k8s.io/kube-controller-manager/config/v1alpha1/types.go]: @liggitt Ugh, do we do names wrong (as it's done here) in literally every config type?

Comment (liggitt) [reply staging/src/k8s.io/kube-controller-manager/config/v1alpha1/types.go]: do you mean not having json tags? this component doesn't yet read config from file, so we are free to rename fields, move them around, etc, prior to adding json tags.

that's on the roadmap to getting the components using external versioned config

Comment (lavalamp) [reply staging/src/k8s.io/kube-controller-manager/config/v1alpha1/types.go]: This is showing up as an API violation. I'd like to not let anything in that adds to the violation file. Reasonable or excessive request?

Comment (liggitt) [reply staging/src/k8s.io/kube-controller-manager/config/v1alpha1/types.go]: that requires adding json tags to the file prior to actually thinking through what the serialized names should be. we didn't want to do that. a file with no json tags is clearly not intended to be serialized yet

Comment (lavalamp) [reply staging/src/k8s.io/kube-controller-manager/config/v1alpha1/types.go]: The tool is complaining that concurrentServiceAccountSyncs != ConcurrentServiceAccountSyncs; if the json tag were added, the tool would be happy. I honestly think the tool is kind of right.

Comment (liggitt) [reply staging/src/k8s.io/kube-controller-manager/config/v1alpha1/types.go]: understood, but we did not want to add json tags before we intended the types to be serialized to files, because it implies they are serializable.

**Comment by andybradshaw** [comment]
@lavalamp @liggitt Happy working on whatever you guys decide on for the path forward here. That being said, I'm not sure what that is at the moment...

1. I'm not sure making the comment name and the struct name match will fix the error, the linter is complaining about fields which are [missing the json tag](https://github.com/kubernetes/kube-openapi/blob/72693cb1fadd73ae2742f6fe29af77d1aecdd8cd/pkg/generators/rules/names_match.go#L97). Note the [somewhat confusing example](https://github.com/kubernetes/kube-openapi/blob/72693cb1fadd73ae2742f6fe29af77d1aecdd8cd/pkg/generators/rules/names_match.go#L74), which is meant to show that using a tag of `json:""` will satisfy this rule. However, I do find the naming inconsistency annoying ( https://github.com/kubernetes/kubernetes/pull/70565).
1. Making the tool skip these types might get a little wonky because of how coupled [generation and validation](https://github.com/kubernetes/kube-openapi/blob/72693cb1fadd73ae2742f6fe29af77d1aecdd8cd/pkg/generators/rules/names_match.go#L74) are... If we split those up, maybe adding a tag (`k8s:openapi-gen-lint=false`?) to skip the structs, either individually or at the package level, would work.
1. Would adding a placeholder tag of `-` be enough to indicate to consumers that attempting to do serialization stuff probably isn't going to do what you want? The presence of a tag, even if the field is ignored, could be confusing.

**Comment by lavalamp** [comment]
If a placeholder of '-' makes the linter happy, I think that makes it
pretty clear to others that the type doesn't serialize.

**Comment by andybradshaw** [comment]
I don't really have a notion of who owns what, so I'm going to keep @'ing people... let me know if I should stop.

@liggitt would the proposed change (json tag of `-`) be acceptable to you?

@lavalamp are we holding that change as blocking this one?

**Comment by lavalamp** [comment]
I want to take a pretty hard line about not adding violations starting asap; I'm willing to let this one in as long as we (i.e. @liggitt) all agree that the next change has to fix this.

**Comment by liggitt** [comment]
> @liggitt would the proposed change (json tag of `-`) be acceptable to you?

sure, that's fine as well

**Comment by andybradshaw** [comment]
@liggitt okay, I can make that change as part of [the comment clean up PR](https://github.com/kubernetes/kubernetes/pull/70565) or a separate PR if that would be preferable

**Comment by lavalamp** [comment]
Either way works for me, thanks!

**Pull request closed without merging**
